### PR TITLE
tests: change default memory request for VM/DV factory

### PR DESCRIPTION
**Pull request by akalenyu**
### What this PR does
1Gi is a LOT, especially since most of the calls are for a tiny cirros/alpine VMs, with a few exceptions.
This triggers CI stress:
```
0/2 nodes are available: 1 Insufficient memory
```
https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12023/pull-kubevirt-check-tests-for-flakes/1812438984626802688/artifacts/k8s-reporter-parallel/2/1_vmis.log

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/libstorage/datavolume.go [hunk 1]
```diff
@@ -229,7 +229,8 @@ func GoldenImageRBAC(namespace string) (*rbacv1.Role, *rbacv1.RoleBinding) {
 func RenderVMIWithDataVolume(dvName, ns string, opts ...libvmi.Option) *v1.VirtualMachineInstance {
 	defaultOptions := []libvmi.Option{
 		libvmi.WithDataVolume("disk0", dvName),
-		libvmi.WithResourceMemory("1Gi"),
+		// This default can be optimized further to 128Mi on certain setups
+		libvmi.WithResourceMemory("256Mi"),
```
Comment (alromeros) [inline tests/libstorage/datavolume.go:233]: Nice, thanks for addressing this!

**Review by alromeros — COMMENTED** [review]
/lgtm

**Comment by akalenyu** [comment]
@jean-edouard hey, do you think we need a whole 1Gi here? https://github.com/kubevirt/kubevirt/blob/e8931cc2f0bfaa6071483a4bbddac9879640c707/tests/vm_state_test.go#L122-L129

**Comment by jean-edouard** [comment]
> @jean-edouard hey, do you think we need a whole 1Gi here?
> 
> https://github.com/kubevirt/kubevirt/blob/e8931cc2f0bfaa6071483a4bbddac9879640c707/tests/vm_state_test.go#L122-L129

We need whatever Fedora needs... (or create an alpine/cirros image that includes the TPM tools and hexdump)

**Comment by akalenyu** [comment]
> > @jean-edouard hey, do you think we need a whole 1Gi here?
> > https://github.com/kubevirt/kubevirt/blob/e8931cc2f0bfaa6071483a4bbddac9879640c707/tests/vm_state_test.go#L122-L129
> 
> We need whatever Fedora needs... (or create an alpine/cirros image that includes the TPM tools and hexdump)

That's fine, Fedora needs 512Mi

**Review by dhiller — COMMENTED** [review]
/approve

Thank you @akalenyu , sounds reasonable! One ask I have is, could you take a look that there's not a flakiness that has been introduced by the changes here?

https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12367/pull-kubevirt-check-tests-for-flakes/1813482595388755968

**Comment by akalenyu** [comment]
> /cherrypick release-1.3

nvm not needed

**Pull request merged**